### PR TITLE
Support for Mobu 2019

### PR DESCRIPTION
**Pull request by manneohrstrom**
- This adds support for mobu 2019 and cleans up the engine payload.
- This version of the engine is not compatible with versions before 2014.
- Removes binary files (qt/pyside distros), thereby reducing the size of the engine drastically.
- Engine now requires core v0.18+
- This is a replacement for #8
- Adds context switching support, as suggested in #7
- Updates to 0.18 logging
- removes deprecated config params `debug_logging` and `use_sgtk_as_menu_name`

### CODE REVIEW

File: engine.py [hunk 1]
```diff
@@ -39,8 +40,13 @@ def tank_mobu_exception_trap(ex_cls, ex, tb):
 
     # now output it    
     try:
-        from PySide import QtGui
```
Comment (thebeeland) [inline engine.py]: [nitpick] Should we also change references to `tank` over to `sgtk` while we're doing cleanup/modernization work in here?

File: engine.py [hunk 2]
```diff
@@ -89,61 +102,31 @@ def init_engine(self):
...
 
-        plugin_path = os.path.join(self.disk_location, "resources", pyside_folder, "qt_plugins")
+        plugin_path = os.path.join(
+            self.disk_location, 
+            "resources", 
+            "qt470_win64_vs2010", 
+            "qt_plugins"
+        )
 
         self.log_debug(
```
Comment (thebeeland) [inline engine.py]: [minor] Were you planning to update all of the legacy log_* calls? If so, missed this one! If not, ignore me!

File: engine.py [hunk 3]
```diff
@@ -173,55 +156,64 @@ def _get_dialog_parent(self):
...
+            formatter = logging.Formatter("Debug: Shotgun %(basename)s: %(message)s")
+        else:
+            formatter = logging.Formatter("Shotgun %(basename)s: %(message)s")
 
-    def log_warning(self, msg):
-        msg = "Shotgun Warning: %s" % msg
-        print msg
+        msg = formatter.format(record)
 
+        # Select Maya display function to use according to the logging record level.
```
Comment (thebeeland) [inline engine.py]: s/Maya/MoBu/

File: engine.py (hunk 1 above)
Comment (manneohrstrom) [reply engine.py]: cool! will do!

File: engine.py (hunk 2 above)
Comment (manneohrstrom) [reply engine.py]: good point! i didn't plan on doing that. but i should :)

File: engine.py (hunk 3 above)
Comment (manneohrstrom) [reply engine.py]: fixed!

File: engine.py (hunk 1 above)
Comment (manneohrstrom) [reply engine.py]: done!

**Pull request merged**